### PR TITLE
refactor(ingest): Auto report workunits

### PR DESCRIPTION
**Pull request by asikowitz**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/api/source.py [hunk 1]
```diff
@@ -118,6 +129,9 @@ class TestConnectionReport(Report):
 WorkUnitType = TypeVar("WorkUnitType", bound=WorkUnit)
 ExtractorConfig = TypeVar("ExtractorConfig", bound=ConfigModel)
 
+WorkUnitProcessor = Callable[[Iterable[WorkUnitType]], Iterable[WorkUnitType]]
+MetadataWorkUnitProcessor = WorkUnitProcessor[MetadataWorkUnit]
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/api/source.py:133]: is this used anywhere? if not, let's remove until it is used

File: metadata-ingestion/src/datahub/ingestion/api/source.py [hunk 2]
```diff
@@ -155,10 +169,6 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> "Source":
         # can't make this method abstract.
         raise NotImplementedError('sources must implement "create"')
 
-    @abstractmethod
-    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
-        pass
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/api/source.py:160]: this change doesn't look right

**Review by hsheth2 — APPROVED** [review]
have some comments on metadata-ingestion/src/datahub/ingestion/api/source.py

otherwise lgtm once the tests pass

File: metadata-ingestion/src/datahub/ingestion/api/source.py (hunk 2 above)
Comment (asikowitz) [reply metadata-ingestion/src/datahub/ingestion/api/source.py:160]: Yeah, I gotta do a full pass, was a messy merge

**Pull request merged**